dts/x86/mem.h

```diff
@@ -9,6 +9,9 @@
 #elif defined(CONFIG_SOC_QUARK_D2000)
 #define DT_FLASH_SIZE		__SIZE_K(32)
 #define DT_SRAM_SIZE		__SIZE_K(8)
+#elif defined(CONFIG_SOC_IA32)
+#define DT_FLASH_SIZE		__SIZE_K(4)
```

Comment:
these values are definitely not right.
We enable 8MB of RAM in the emulator.
If XIP is enabled, we use half of RAM for "flash" (really just to test all the XIP copying stuff) and the other half for system RAM. The very first page of memory (0-4K) is left unmapped so that NULL accesses pagefault.